modules/distributor/queue/queue.go

```diff
@@ -0,0 +1,224 @@
+package queue
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/user"
+	"go.uber.org/atomic"
```

Comment:
would prefer moving to the std lib implementation unless there's a reason to use this one.